src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/BrouwerLyddaneMeanLong.cpp

```diff
@@ -0,0 +1,433 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Types/Size.hpp>
+#include <OpenSpaceToolkit/Core/Utilities.hpp>
+
+#include <OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Kepler/BrouwerLyddaneMeanLong.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+namespace orbit
+{
+namespace models
+{
+namespace kepler
+{
+
+using ostk::core::types::Size;
+using ostk::core::types::Integer;
+
+using EarthGravitationalModel = ostk::physics::environment::gravitational::Earth;
+
+BrouwerLyddaneMeanLong::BrouwerLyddaneMeanLong(
+    const Length &aSemiMajorAxis,
+    const Real &anEccentricity,
+    const Angle &anInclination,
+    const Angle &aRaan,
+    const Angle &anAop,
+    const Angle &aMeanAnomaly
+)
+    : BrouwerLyddaneMean(aSemiMajorAxis, anEccentricity, anInclination, aRaan, anAop, aMeanAnomaly)
+{
+}
+
+COE BrouwerLyddaneMeanLong::toCOE() const
+{
+    const auto mod = [](const Real &value, const Real &bound) -> Real
+    {
+        return value - bound * std::floor(value / bound);
+    };
+
+    // TBM: Make configurable?
+    const Real equatorialRadius = EarthGravitationalModel::EGM2008.equatorialRadius_.inMeters();
+
+    Integer pseudostate = 0;
+    bool possibleInaccuracyLongWritten = false;
+    bool criticalInclinationWritten = false;
+
+    const Real re = EarthGravitationalModel::EGM2008.equatorialRadius_.inMeters();
+    const Real j2 = 1.082626925638815E-03;
+    const Real j3 = -0.2532307818191774E-5;
+    const Real j4 = -0.1620429990000000E-5;
+    const Real j5 = -0.2270711043920343E-6;
```

Comment:
Side question: Don't we need J2_2 as well?

Reply:
I'd prefer just having the exact same values as GMAT here, the fewer discrepancies the better.
re is being pulled because it's just a scaling factor for SMA.

GMAT doesn't use J2_2...

Reply:
> GMAT doesn't use J2_2...

Hum interesting. We might not need it but I am mentioning it because it appears in on of the references https://space.stackexchange.com/questions/22151/whats-a-brouwer-lyddane-mean-semi-major-axis-or-any-other-for-an-orbit-in-a-l

<img width="720" alt="Screen Shot 2023-10-13 at 3 17 16 PM" src="https://github.com/open-space-collective/open-space-toolkit-astrodynamics/assets/43561754/c4df25e7-6212-4cc2-9488-6886fc31971e">

Reply:
Hum, or maybe it's only relevant for >= J6 🤔

Reply:
Hmmmmm interesting. Yeah it might be for >= J6. I don't have a copy of Vallado otherwise I'd double check. But it looks like Orekit has a similar implementation https://github.com/CS-SI/Orekit/blob/develop/src/main/java/org/orekit/propagation/analytical/BrouwerLyddanePropagator.java (J2-J5).